### PR TITLE
Do not try to switch file/buffer when jumping inside same file

### PR DESCRIPTION
**Pull request by Hi-Angel**
The vast majority of jumps will end up within the current opened
file/buffer, simply because usually users do more movement inside a
buffer than between multiple buffers.

There is even a special case when evil-jumps-cross-buffers is set to
nil, which causes jumps never ever switch buffers.

So, let's do a small optimization, and not try to call find-file or
switch-to-buffer if we know it is unnecessary.

As a side-effect, this should also help to users to stumble less upon
https://github.com/emacs-evil/evil/issues/1504 bug (it has its own
upstream fix, but until that landed, this small optimization is known to
help as well).

### CODE REVIEW

**Comment by Hi-Angel** [comment]
I assume the v24.5 tests fail is unrelated, because looking at the recent PRs, all of them has that.

File: evil-jumps.el [hunk 1]
```diff
@@ -156,9 +156,10 @@
                (pos (car place))
                (file-name (cadr place)))
           (setq evil--jumps-jumping t)
-          (if (string-match-p evil--jumps-buffer-targets file-name)
-              (switch-to-buffer file-name)
-            (find-file file-name))
+          (when (not (string= current-file-name file-name))
```
Comment (tomdl89) [inline evil-jumps.el:159]: `(unless (string= current-file-name file-name)`
tends to be the idiom

**Comment by tomdl89** [comment]
Indeed the v24.5 tests fail is unrelated. We need to get around to fixing that. Thanks for the PR. Looks good.

File: evil-jumps.el (hunk 1 above)
Comment (Hi-Angel) [reply evil-jumps.el]: Nice idea, thank you! Done!

**Pull request merged**